code/modules/research/bepis.dm

```diff
@@ -0,0 +1,248 @@
+	gauss_minor = (gaussian(minor_threshold, std) - negative_cash_offset)	//And this is the threshold to instead get a minor prize.
+	gauss_real = (gaussian(banked_cash, std*inaccuracy_percentage) + positive_cash_offset)	//this is the randomized profit value that your experiment expects to give.
+	say("Real, [gauss_real]. Minor, [gauss_minor]. Major, [gauss_major].")
+	if(gauss_real >= gauss_major) //Major Success.
+		success_type = 2
+		return
+	else if(gauss_real >= gauss_minor) //Minor Success.
+		success_type = 1
+		return
+	else if(gauss_real <= -1)
+		success_type = 3 //Critical Failure.
+		return
+	success_type = 0 //Neutral Failure.
+	return
+
+/obj/machinery/rnd/bepis/update_icon_state()
+	if(is_operational() && (chamber_status < 2))
+		chamber_status = 1
+	if(((powered == FALSE) && (banked_cash == 0)) || (!is_operational()))
+		chamber_status = 0
+	if((powered == TRUE) && (banked_cash > 0))
+		chamber_status = 2
+	else if ((powered == FALSE) && (banked_cash > 0))
+		chamber_status = 3
+	switch(chamber_status)
+		if(0)
+			icon_state = "chamber"
+		if(1)
+			icon_state = "chamber_active"
+		if(2)
+			icon_state = "chamber_active_loaded"
+		if(3)
+			icon_state = "chamber_loaded"
+
+
+/obj/machinery/rnd/bepis/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "bepis", name, 500, 400, master_ui, state)
+		ui.open()
+	if(ishuman(user))
+		H = user
+		Card = H.get_idcard(TRUE)
+		update_icon_state()
+	RefreshParts()
+	update_icon_state()
+
+/obj/machinery/rnd/bepis/ui_data()
+	var/list/data = list()
+	var/storedcash = banked_cash
+
+	data["amount"] = banking_amount
+	data["stored_cash"] = storedcash
+	data["mean_value"] = major_threshold
+	data["error_name"] = error_cause
+	data["power_saver"] = power_saver
+	data["accuracy_percentage"] = inaccuracy_percentage * 100
+	data["positive_cash_offset"] = positive_cash_offset
+	data["negative_cash_offset"] = negative_cash_offset
+	data["manual_power"] = powered ? FALSE : TRUE
+	return data
+
+/obj/machinery/rnd/bepis/ui_act(action,params)
+	if(..())
+		return
+	switch(action)
+		if("deposit_cash")
+			if(powered == FALSE)
+				return
+			depositcash()
+			update_icon_state()
+		if("begin_experiment")
+			var/turf/dropturf = get_turf(pick(view(1,src)))
+			if(!dropturf) //Check to verify the turf exists and the reward isn't lost somehow.
+				dropturf = drop_location()
+			if(powered == FALSE)
+				return
+			calcsuccess()
+			if(banked_cash == 0)
+				say("Please deposit funds to begin testing.")
+				return
+			use_power(10000 * power_saver) //This thing should eat your APC battery if you're not careful.
+			//For assigning major rewards, the tech disk will have a random tech with the bepis_tech variable true, check _techweb.dm
+			if(success_type == 2)
```

Comment:
This success_type variable seems completely temporary too, return value from calc success instead.